pkg/internal/computeresources/transformer_test.go

```diff
@@ -968,49 +968,774 @@ func TestTransformerMultipleContainer(t *testing.T) {
+				Name:  "step-foo",
+				Image: "baz",
+				Resources: corev1.ResourceRequirements{
+					Limits: createResourceList(2),
+				},
+			}},
+		},
+		wantInit: []corev1.Container{{
+			Resources: corev1.ResourceRequirements{
+				Requests: createResourceList(3),
+			},
+		}},
+	}, {
+		description: "multiple containers with requests and limits",
+		podspec: corev1.PodSpec{
+			InitContainers: []corev1.Container{{
+				Name:  "bar",
+				Image: "foo",
+			}},
+			Containers: []corev1.Container{{
+				Name:  "step-foo",
+				Image: "baz",
+				Resources: corev1.ResourceRequirements{
+					Requests: createResourceList(1),
+					Limits:   createResourceList(5),
+				},
+			}, {
+				Name:  "step-foo",
+				Image: "baz",
+				Resources: corev1.ResourceRequirements{
+					Requests: createResourceList(2),
+					Limits:   createResourceList(2),
+				},
+			}},
+		},
+		wantInit: []corev1.Container{{
+			Resources: corev1.ResourceRequirements{
+				Requests: createResourceList(2),
+				Limits:   createResourceList(5),
+			},
+		}},
+	}, {
+		description: "multiple containers with requests and limits; multiple init containers",
+		podspec: corev1.PodSpec{
+			InitContainers: []corev1.Container{{
+				Name:  "bar",
+				Image: "foo",
+			}, {
+				Name:  "baz",
+				Image: "qux",
+			}},
+			Containers: []corev1.Container{{
+				Name:  "step-foo",
+				Image: "baz",
+				Resources: corev1.ResourceRequirements{
+					Requests: createResourceList(1),
+					Limits:   createResourceList(5),
+				},
+			}, {
+				Name:  "step-foo",
+				Image: "baz",
+				Resources: corev1.ResourceRequirements{
+					Requests: createResourceList(2),
+					Limits:   createResourceList(2),
+				},
+			}},
+		},
+		wantInit: []corev1.Container{{
+			Resources: corev1.ResourceRequirements{
+				Requests: createResourceList(2),
+				Limits:   createResourceList(5),
+			},
+		}, {
+			Resources: corev1.ResourceRequirements{
+				Requests: createResourceList(2),
+				Limits:   createResourceList(5),
+			},
+		}},
+	}} {
+		t.Run(tc.description, func(t *testing.T) {
+			pod := corev1.Pod{Spec: tc.podspec}
+			withConfig := enableInitContainerResources(t)
+			ctx := withConfig(context.Background())
+			got := transformPodComputeResources(ctx, &pod, nil)
+			cmpContainers(t, tc.wantInit, got.Spec.InitContainers)
+		})
 	}
+}
 
-	// diff containers
-	if len(want.Containers) != len(got.Containers) {
-		t.Errorf("Expected %d containers, got %d", len(want.Containers), len(got.Containers))
-	} else {
-		for i, c := range got.Containers {
-
-			// compare name only if present in "want" so we can be sure which container gets which resources
-			if want.Containers[i].Name != "" {
-				if d := cmp.Diff(want.Containers[i].Name, c.Name); d != "" {
-					t.Errorf("Diff containers[%d] %s", i, diff.PrintWantGot(d))
-				}
-			}
-
-			if d := cmp.Diff(want.Containers[i].Resources, c.Resources, resourceQuantityCmp); d != "" {
-				t.Errorf("Diff containers[%d] %s", i, diff.PrintWantGot(d))
-			}
+/*
```

Comment:
Were these tests meant to be commented out?

Reply:
😳 no lol I'll fix